### PR TITLE
WIP: Added support for compiling with vendored dependencies inside of docker

### PR DESCRIPTION
**Pull request by kolaente**
I've had an issue where I had dependencies inside the `vendor` folder which are not available when directly using `go get`, so I was using vendored mode. However, it seems xgo still downloads all depencies and just ignores whatever there is inside of the `vendor/`. This pr hopefully fixes that.

### CODE REVIEW

**Comment by kolaente** [comment]
I'll try to add some more test cases for these.

File: xgo.go [hunk 1]
```diff
@@ -368,6 +370,24 @@ func compileContained(config *ConfigFlags, flags *BuildFlags, folder string) err
...
+
+		// Check whether it has a vendor folder, and if so, use it
+		absRepository, err := filepath.Abs(config.Repository)
+		if err != nil {
+			log.Fatalf("Failed to locate requested module repository: %v.", err)
+		}
+		vendorPath := absRepository + "/vendor"
+		vendorfolder, err := os.Stat(vendorPath)
+		if !os.IsNotExist(err) && vendorfolder.Mode().IsDir() {
+			env = append(env, "FLAG_MOD=vendor")
```
Comment (naiba) [inline xgo.go]: Is that should be `GOFLAGS=-mod=vendor`?

**Comment by kolaente** [comment]
@naiba thanks, but it's not finished yet :slightly_smiling_face:

**Comment by kolaente** [comment]
mhh looks like since we updated xgo to use go 1.12.6 this does not build anymore.

**Comment by shilch** [comment]
Bump @techknowlogick

**Comment by kolaente** [comment]
@shilch I'd like to add some test cases for this, but would like to fix #48 first (which is broken atm)

**Comment by shilch** [comment]
Thanks for the response @kolaente. What I do right now to use the vendor folder is mounting it into the /go/src directory. This way, xgo doesn't download the dependencies.

**Comment by techknowlogick** [comment]
Closing this as we use this works(?)-ish with Gitea, although as mentioned we need to focus on testing.

**Pull request closed without merging**